src/muxer.ts

```diff
@@ -419,20 +433,25 @@ export class Muxer<T extends Target> {
 		data: Uint8Array,
 		type: 'key' | 'delta',
 		timestamp: number,
+		compositionTimeOffset: number,
 		duration: number,
 		meta?: EncodedVideoChunkMetadata | EncodedAudioChunkMetadata
 	) {
-		let timestampInSeconds = timestamp / 1e6;
+		let ptsInSeconds = timestamp / 1e6;
+		let dtsInSeconds = (timestamp - compositionTimeOffset) / 1e6;
```

Comment:
Well, everywhere else that we present `timestamp` and `compositionTimeOffset` (including to the user), `timestamp` is the presentation time. And since composition time offset = PTS-DTS, then also DTS=PTS-CTO. It feels confusing to me to suddenly switch the meaning of `timestamp` here from PTS to DTS. Or am I misunderstanding what you're trying to say?

Reply:
No, you understand correctly. My confusion stemmed from the fact that compositionTimeOffset sounds like we need to apply an offset to the timestamp to determine the composition time. It sounds like compositionTime is a distinct concept from timestamp. That's why I assumed that timestamp referred to the decode time instead.

Of course, if it doesn't, then informationally they are still identical. If you give me the composition time and the composition time offset, you can deduce DTS using PTS - CTO. However, I feel it is more natural to add offsets than to subtract them, so perhaps `decodeTimeOffset` is better? This implies that the passed timestamp is, in fact, the presentation time and that the decode time can differ. Then you can reach DTS using PTS + DTO.

I guess this is a semantic nitpick. I just (at least internally) view the decode time as the "main" timestamp, the timestamp that the chunks in the file are sorted by, the timestamps that are monotonically increasing, and the timestamps that chunks are divided by. May I ask, when you said

> Well, everywhere else that we present timestamp and compositionTimeOffset (including to the user), timestamp is the presentation time.

which instances you were referring to?

The encoder that you're using that generates B-frames; what its take on this? Does it output `{ presentationTime, CTO }` or `{ decodeTime, CTO }`?

Reply:
> > Well, everywhere else that we present timestamp and compositionTimeOffset (including to the user), timestamp is the presentation time.
> which instances you were referring to?

I was talking mainly about `addVideoChunk`/`addVideoChunkRaw`, which take `timestamp` and `compositionTimeOffset`. While this repo's README doesn't describe `timestamp` as being related to presentation, composition, or decoding, the WebCodecs standard does appear to associate the `timestamp` of various entities (e.g. `EncodedAudioChunk`, `EncodedVideoChunk`, etc.) with the presentation of the media (e.g. https://www.w3.org/TR/webcodecs/#encodedvideochunk-interface). So my motivation was to keep the definition of `timestamp` in `addVideoChunk`/`addVideoChunkRaw` consistent with the WebCodecs interfaces (meaning it's a presentation timestamp).

Having said that, since `#createSampleForTrack` is internal, we can do whatever you want; we could keep `timestamp` and `compositionTimeOffset`; we could switch to `decodeTimestamp` and `compositionTimeOffset`; we could use `presentationTimestamp` and `decodeTimestamp`... the choice is yours.

> The encoder that you're using that generates B-frames; what its take on this?

Well, it's my own company's proprietary software, so its (my?) take is biased. Currently, the frames are presented over a web socket with separate PTS and DTS; internally in the software, they're passed around with a `Timestamp` (PTS) and a `Delay` (CTO).

Reply:
> However, I feel it is more natural to add offsets than to subtract them, so perhaps decodeTimeOffset is better?

I don't necessarily disagree with your logic, but I think that "composition time offset" is a term that is used in standards. For example, it appears several times in ISO/IEC 14496-12.

Reply:
If the WebCodec specs refer to `timestamp` as the presentation time, that's enough of a reason for me - convinced! Then let's keep it like this.